dGame/dComponents/TriggerComponent.cpp

```diff
@@ -0,0 +1,214 @@
+#include "TriggerComponent.h"
+
+#include "dZoneManager.h"
+#include "LUTriggers.h"
+#include "eTriggerCommandType.h"
+
+#include "MissionComponent.h"
+#include "PhantomPhysicsComponent.h"
+
+#include "CDMissionTasksTable.h"
+
+
+TriggerComponent::TriggerComponent(Entity* parent, const std::string triggerInfo): Component(parent) {
+	m_Parent = parent;
+	m_Trigger = nullptr;
+
+	std::stringstream ss(triggerInfo);
+	std::vector<std::string> tokens;
+	std::string token;
+	while (std::getline(ss, token, ':')) {
+		tokens.push_back(token);
+	}
+
+	uint32_t sceneID;
+	GeneralUtils::TryParse<uint32_t>(tokens[0], sceneID);
+	uint32_t triggerID;
+	GeneralUtils::TryParse<uint32_t>(tokens[1], triggerID);
+
+	if (m_Trigger) {
+		delete m_Trigger;
+		m_Trigger = nullptr;
+	}
+
+	m_Trigger = dZoneManager::Instance()->GetZone()->GetTrigger(sceneID, triggerID);
+
+	if (!m_Trigger) m_Trigger = new LUTriggers::Trigger();
+}
+
+void TriggerComponent::TriggerEvent(eTriggerEventType event, Entity* optionalTarget) {
+	if (m_Trigger && m_Trigger->enabled) {
+		for (LUTriggers::Event* triggerEvent : m_Trigger->events) {
+			if (triggerEvent->id == event) {
+				for (LUTriggers::Command* command : triggerEvent->commands) {
+					HandleTriggerCommand(command, optionalTarget);
+				}
+			}
+		}
+	}
+}
+
+void TriggerComponent::HandleTriggerCommand(LUTriggers::Command* command, Entity* optionalTarget) {
+	auto argArray = ParseArgs(command->args);
+
+	// determine targets
+	std::vector<Entity*> targetEntities = GatherTargets(command, optionalTarget);
+
+	// if we have no targets, then we are done
+	if (targetEntities.empty()) return;
+
+	for (Entity* targetEntity : targetEntities) {
+		if (!targetEntity) continue;
+
+		switch (command->id) {
```

Comment:
Once again, thank you mac for the amazing warning

It does have a point though where we could at the very least log that we got an invalid trigger event this far for debugging purposes?